heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/MetricsCacheManager.java

```diff
@@ -0,0 +1,258 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.twitter.heron.metricscachemgr;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.twitter.heron.common.basics.Constants;
+import com.twitter.heron.common.basics.NIOLooper;
+import com.twitter.heron.common.basics.SingletonRegistry;
+import com.twitter.heron.common.basics.SysUtils;
+import com.twitter.heron.common.config.SystemConfig;
+import com.twitter.heron.common.network.HeronSocketOptions;
+import com.twitter.heron.common.utils.logging.ErrorReportLoggingHandler;
+import com.twitter.heron.common.utils.logging.LoggingHelper;
+import com.twitter.heron.metricscachemgr.metricscache.MetricsCache;
+import com.twitter.heron.metricsmgr.MetricsSinksConfig;
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+import com.twitter.heron.spi.common.ClusterConfig;
+import com.twitter.heron.spi.common.ClusterDefaults;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.Context;
+import com.twitter.heron.spi.common.Defaults;
+import com.twitter.heron.spi.common.Keys;
+import com.twitter.heron.spi.statemgr.IStateManager;
+import com.twitter.heron.spi.utils.ReflectionUtils;
+
+//import com.twitter.heron.spi.common.ConfigDefaults;
+
+//import com.twitter.heron.scheduler.TopologyRuntimeManagementException;
+
+/**
+ * main entry for metrics cache manager
+ */
+public class MetricsCacheManager {
+  // logger
+  private static final Logger LOG = Logger.getLogger(MetricsCacheManager.class.getName());
+
+  // Pre-defined value
+  private static final String METRICS_CACHE_HOST = "0.0.0.0";
+  private static final String METRICS_CACHE_COMPONENT_NAME = "__metricscachemgr__";
+  private static final int METRICS_CACHE_INSTANCE_ID = -1;
+
+  // accepts messages from sinks
+  private MetricsCacheManagerServer metricsCacheManagerServer;
+  // The looper drives MetricsManagerServer
+  private NIOLooper metricsCacheManagerServerLoop;
+
+  // respond to query requests
+  private MetricsCacheManagerHttpServer metricsCacheManagerHttpServer;
+
+  // process identification
+  private String topologyName;
+  private String metricsmgrId;
+
+  // store
+  private MetricsCache metricsCache;
+
+  // holds all the config read
+  private Config config;
+
+  // location for state mgr
+  private TopologyMaster.MetricsCacheLocation metricscacheLocation;
+
+  /**
+   * Constructor
+   *
+   * @param topologyName topology name
+   * @param serverHost server host
```

Comment:
Until we have a use case where we need to bind to a specific ip, let's not support this. It just adds to the explosion of params required to be managed and I don't think we support it for other processes.